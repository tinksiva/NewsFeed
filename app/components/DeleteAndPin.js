import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import { memo, useCallback } from "react";

//scale function is used to make the ui responsive
import { scale, scaleFont } from "../utils/scale";
import { colors } from "../themes/Colors";

function DeleteAndPin(props) {
  const {
    id,
    deleteItem,
    pinItem,
    unPinItem,
    isPinned = false,
  } = props;

  //Component that displays the delete and pin options
  const pinUnpin = useCallback(() => {
    if (isPinned) {
      unPinItem();
    } else {
      pinItem(id);
    }
  }, [])
  const deleteNews = useCallback(() => {
    deleteItem(id, isPinned);
  }, [])
  return (
    <View style={deletePinStyles.rightSwipeItemsContainer}>
        <TouchableOpacity
          style={[
            deletePinStyles.deletePinContainer,
            deletePinStyles.deleteColor,
          ]}
          onPress={deleteNews}
        >
          <Text style={deletePinStyles.deletePinText}>{"Delete"}</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[deletePinStyles.deletePinContainer, deletePinStyles.pinColor]}
          onPress={pinUnpin}
        >
          <Text style={deletePinStyles.deletePinText}>
            {isPinned ? "UnPin" : "Pin"}
          </Text>
        </TouchableOpacity>
      </View>
  );
}



const deletePinStyles = StyleSheet.create({
  deletePinContainer: {
    alignSelf: "center",
    height: "50%",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    paddingHorizontal: "2%",
  },
  deleteColor: { backgroundColor: colors.deleteColor },
  pinColor: { backgroundColor: colors.pinColor },
  deletePinText: { alignSelf: "center", color: colors.white },
  rightSwipeItemsContainer: {
    height: scale(135),
    marginVertical: "2%",
    borderRadius: scale(10),
    overflow: "hidden",
  },
});

export default memo(DeleteAndPin);
