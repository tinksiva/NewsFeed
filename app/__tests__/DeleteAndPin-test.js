jest.useFakeTimers()
import 'react-native';
import React from 'react';
import DeleteAndPin from "../components/DeleteAndPin";

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <DeleteAndPin />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});