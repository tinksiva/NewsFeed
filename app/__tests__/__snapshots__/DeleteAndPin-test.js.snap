// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "borderRadius": 21.428571428571427,
      "height": 289.2857142857143,
      "marginVertical": "2%",
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#d11a2a",
        "height": "50%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": "2%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "white",
        }
      }
    >
      Delete
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#3D7BED",
        "height": "50%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": "2%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "white",
        }
      }
    >
      Pin
    </Text>
  </View>
</View>
`;
